### DISCUSSION TITLE
Есть потребность в публичном доступе к методу setupMapChildren

### DISCUSSION THREAD

**Discussion by dongaber** [yandex-maps-unofficial/vue-yandex-maps#247] [Q&A (2.x)] [answered]
Я сделал кастомную реализацию entity: export class YMapCustomEntity extends ymaps3.YMapGroupEntity<{}>
Хотел использовать как Vue компонент, но столкнулся с отсутствием доступа к функции: setupMapChildren, для создания кастомных компонентов. Нужно ли для этого делать публичный доступ к функции или я что то упускаю?

**Comment by daniluk4000** [comment]
А подскажите, какой кейс? Может компонент лучше интегрировать в либу? Или всё таки хочется, чтобы функция была? В целом я соглашусь, что так будет больше гибкости для тех, кто хочет комбинировать либу и Яндексовые компоненты

Мне в целом не жалко, просто тогда её придется документировать) Если хочется - закиньте Feature Request в ишьюсы и я сделаю (но не в ближайшую неделю)

**Comment by daniluk4000** [comment]
А вообще вы еще можете через map.value.addChild (и по аналогии с другими инстансами) делать. setupMapChildren всё таки больше внутренняя, у вас есть v-model и апишка Яндекса уже в нем

**Comment by dongaber** [comment]
Мой кейс как раз нужн для рисования полигона на карте и получения его координат. Вообще наверное смогу сделать через map.value.addChild, сейчас попробую.

**Reply by daniluk4000** [reply]
Ага, давайте. По идее так должно прокатить. А setupMapChildren скорее для взаимодействия внутренних оберток между собой.

**Reply by dongaber** [reply]
Да и еще не могли бы вы взглянуть на реализацию, это не конечный вариант и хотелось бы узнать ваше мнение по коду, может я зря это делаю таким способом.
https://codefile.io/f/wFkDzlE6Oo

**Reply by daniluk4000** [reply, marked as answer]
@NickRisinger просто пишите все эти компоненты библиотеки в теле <yandex-map> и всё. Зачем создавать YMapCustomEntity в принципе? Всё, что вы в нём используете, можно использовать в Vue и реактивно

**Reply by dongaber** [reply]
хм, видимо сам усложнил себе работу, сейчас попробую сделать все средствами библиотеки

**Reply by daniluk4000** [reply]
Возможно, к этому приводит чтение документации Яндекса) Но ладно, без негатива в их сторону

Пишите, если потребуется подсказка или помощь)

**Reply by dongaber** [reply]
Думаю, да, их документация божественна))
Спасибо за ответ

**Reply by dongaber** [reply]
Сделал реализацию средствами библиотеки, как вы и посоветовали, и о чудо все работает, даже лучше чем было, еще раз спасибо!

**Reply by daniluk4000** [reply]
Всегда рад!
